src/libraries/System.Private.CoreLib/src/System/Runtime/Intrinsics/Vector128.cs

```diff
@@ -1369,6 +1370,11 @@ public static unsafe Vector128<ulong> Create(Vector64<ulong> lower, Vector64<ulo
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
         public static unsafe Vector128<byte> CreateScalar(byte value)
         {
+            if (AdvSimd.IsSupported)
+            {
+                return AdvSimd.Insert(Vector128<byte>.Zero, 0, value);
```

Comment:
We'll need to special-case `CreateScalarUnsafe` since the upper bits don't have to be zeroed for it.

Reply:
Yep, this work is tracked by https://github.com/dotnet/runtime/issues/34485